### ISSUE TITLE
Implement a command line tool to transcribe a set of files with cubic server

### ISSUE THREAD

**Issue by happyalu** [cobaltspeech/cubic-cli#2]
Implement a client in go that uses the go-cubic package and sends sets of files to cubic server to get them transcribed.

**Comment by jhollowayj** [comment]
Here's my review of requirements:

Flags: 
```
--server [string] address of cubic server (I'll assume insecure connections for now.
--models [bool] Simply returns the list of availabe models and quits
--version [bool] Simply returns the cubic svr model (and maybe the client version?) and quits
--file [string] Path to single file to transcribe.  Mutually Exclusive with --listFile
--listFile [string] Path to text file containing list of `utteranceID \t path/to/file`.  Mutually exclusive with --file.
--nConcurrent [int]  Number of concurrent requests to send to cubicsvr.  Only valid with --listFile.
~~--realTime [bool] If true, sleeps will be introduced to delay the streaming.  Simulates a "real time" use case.~~
--outputFile [string] Path to file where output should be sent to.  Empty flag will result in stdout.  Path is checked for existing file at start.  If file exists, the program will warn and exit unless --overwriteOutputFile is set. 
--overwriteOutputFile [bool] If set and outputFile exists already, the warning will still be made, but the program will continue, and all existing content will be overwritten.  (no effect if outputFile does not exist or stdout is being used.)
--insecure [bool] disables TLS.  (default it to use TLS)
```

Input format:
* The exact format of the input is subject to change. 
* Will likely be 1x `UttID \t path/to/audio/file` per line.

Output Format:
* Text format only (for now)
* `UttID \t (1-best) transcription\n`
* Utterances will be written to file as they are streamed back in.
* UttIDs maybe be intreweived, but the order of any given UttID's segments will be in order.
* Example:
  * UttID_1 \t output.1
  * UttID_2 \t output.1
  * UttID_1 \t output.2
  * UttID_3 \t output.1
  * UttID_3 \t output.2

Implementation Details:
* Filenames -> inChannel (sorted by file size)
* nConcurrent workers read next filename (<- inChannel), send content to svr in chunks, (sleep), add add results to (->outChannel)
* (<-outChannel) gets sent to output file.

Things _not_ being implemented:
--chunkSize flag
clientside formatting.  results from cubicsvr will be used directly.

------

Future revisions may implement: 
* other output formats, such as JSON or something like VTT or TTLM.
* Metrics output
* Client side load balancing and support for multiple cubicsvrs.
* Flags to support secure connections to cubicsvr.
* clientside integrations with cobalt formatting services. (i.e. fmtSvr<->client<->cubicsvr)

**Comment by happyalu** [comment]
> --address [string] address of cubic server (I'll assume insecure connections for now.

The default should be to use TLS. We should have an`--insecure` flag to disable it.

In terms of the CLI, how about:

```
$ cubic --server addr:port [--insecure] version
$ cubic --server addr:port models
```
```
$ cubic --server addr:port recognize \
    --model-id ID \
    [--workers N] \
    [--realtime] \
    [--output FILE] \
    [--list-file] INPUT
```

**Comment by jhollowayj** [comment]
This issue been implemented in https://github.com/cobaltspeech/cubic-cli.git

**Comment by jhollowayj** [comment]
Closed by #1
